### ISSUE TITLE
FastAPI support

### ISSUE THREAD

**Issue by erm** [erm/mangum#22]
Using the following application code:

```python
from mangum.platforms.aws.adapter import AWSLambdaAdapter
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def read_root():
    return {"hello": "world"}


handler = AWSLambdaAdapter(app)
```

I ran into this error after deploying to AWS:

```shell
[ERROR] AttributeError: module 'typing' has no attribute '_ClassVar'
Traceback (most recent call last):
  File "/var/lang/lib/python3.7/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/var/lang/lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/var/task/asgi.py", line 19, in <module>
    from fastapi import FastAPI
  File "/var/task/fastapi/__init__.py", line 5, in <module>
    from .applications import FastAPI
  File "/var/task/fastapi/applications.py", line 3, in <module>
    from fastapi import routing
  File "/var/task/fastapi/routing.py", line 6, in <module>
    from fastapi import params
  File "/var/task/fastapi/params.py", line 4, in <module>
    from pydantic import Schema
  File "/var/task/pydantic/__init__.py", line 2, in <module>
    from .class_validators import validator
  File "/var/task/pydantic/class_validators.py", line 19, in <module>
    @dataclass
  File "/var/task/dataclasses.py", line 958, in dataclass
    return wrap(_cls)
  File "/var/task/dataclasses.py", line 950, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
  File "/var/task/dataclasses.py", line 801, in _process_class
    for name, type in cls_annotations.items()]
  File "/var/task/dataclasses.py", line 801, in <listcomp>
    for name, type in cls_annotations.items()]
  File "/var/task/dataclasses.py", line 659, in _get_field
    if (_is_classvar(a_type, typing)
  File "/var/task/dataclasses.py", line 550, in _is_classvar
    return type(a_type) is typing._ClassVar
```

**Comment by erm** [comment]
Exact package versions that caused this:

```
dataclasses-0.6
fastapi-0.1.18
mangum-0.6.2
pydantic-0.18.2
starlette-0.9.10
```

**Comment by erm** [comment]
This appears to have just been a packaging issue, running the `pip install -r requirements.txt -t ./build` command would include the `dataclasses` package, but I haven't seen it when installing outside of this case and was able to get FastAPI working with Mangum.

cc @tiangolo

**Comment by tiangolo** [comment]
Thanks for your work on this @erm ! Sorry for the delay, I had this issue/email in the backlog as I wanted it as a reminder to document mangum inside of FastAPI :smile: 

I haven't been able to yet, but hopefully I'll do it soon.

**Comment by erm** [comment]
@tiangolo that's great to hear, thanks!

I haven't had a lot of time lately to do much application testing, so it will be really helpful to see some example usage that I can experiment with to refine the adapter implementation and hopefully work out some generic configuration examples to document here as well.

**Comment by mdavis-xyz** [comment]
I'm interested in deploying FastAPI inside AWS Lambda.

This issue is the only mention I've found online of `from mangum.platforms.aws.adapter import AWSLambdaAdapter`.

Where can I find documentation about this adapter?

I found [magnum on pypi](https://pypi.org/project/magnum/) but that description sounds unrelated to this, and I can't find `AWSLambdaAdapter` mentioned anywhere in the source code.

Issue #1233 mentions [this repo](https://github.com/erm/magnum), which appears to no longer exist.

is the `magnum` project no longer supported?
